### PR TITLE
Fix conformance to Zim (and enable Insert Audio button): wrong attachment directories, linked file not found, missing image link's tags, minsdk 16-18

### PR DESCRIPTION
**Pull request by mehw**
Hello,

This series of patches is to **conform the wikitext format to Zim**:
1. The expression `{{}}` is for images, other types should use `[[]]`.
2. **Notebook's links should be relative to a wiki page's attachment directory**.
3. Files pulled into a Notebook should be placed in a proper attachment directory.
4. Interpret image link's tags: `id`, `width`, `height`, `href`; images can link other files.

As an extra, the **Insert Audio** button is enabled.

**More descriptions about Zim conformance are in the commits and in the code as comments.**

# Zim attachment directories

In `wiki-example.txt`, `[[./video.mkv]]` would be a link to a file residing in the `wiki-example` directory (relative path).  Linking `notebook.zim` would add `[[../notebook.zim]]`, and `[[/home/user/README]]` would be a link to a file outside the Notebook's root (absolute path).

```
/home/user/Notebook            # Notebook directory
      |    |
      |    `- notebook.zim     # marks the root of the Notebook
      |    |
      |    `- wiki-example     # wiki page's attachment directory
      |    |  |
      |    |  `- video.mkv     # wiki-example.txt file attachment
      |    |
      |    `- wiki-example.txt # wiki file written in wiki markup
      |
      `- README                # file outside the Notebook's root
```

# Zim image link's tags

If `{{./image.png?href=.%2Fvideo.mkv&id=anchor-name&width=50}}` is in `wiki-example.txt`, `./` is resolved to the wiki page's attachment directory `wiki-example`, and in Markor you would have something like:

`<a href="/home/user/Notebook/wiki-example/video.mkv">wiki-example.txt</a>`

### CODE REVIEW

**Comment by gsantner** [comment]
Hello @mehw the CI shows some build errors with these PR changes. Can you please check/fix that?

**Comment by mehw** [comment]
@gsantner I'm on it.

**Comment by mehw** [comment]
> Hello @mehw the CI shows some build errors with these PR changes. Can you please check/fix that?

@gsantner I [changed a test to reflect a new behavior](https://github.com/gsantner/markor/pull/2147/commits/d585c7b0b9c9e1972abb09ed9d2a3146f8970a83), this fixes the CI: _assume the current directory as top link if root cannot be determined_.

I thought this as a _fallback_, but also if the user wants to organize a notebook as a relative structure, _without a specific root directory_.

When `notebook.zim` is required to be found (aka `isWikitextDynamicNotebookRootEnabled`) but it doesn't exist, the current page's parent directory (aka **current directory**) _is assumed to be the notebook's root from where to resolve the page's links_.

https://github.com/gsantner/markor/blob/856175fcb0390f3a618cf694e06d9f0b8abb8860/app/src/main/java/net/gsantner/markor/format/wikitext/WikitextLinkResolver.java#L94-L95

**Comment by gsantner** [comment]
Due other PR's edited the same files/things, there are merge conflicts now.

Can you please fix it?

**Comment by mehw** [comment]
> Due other PR's edited the same files/things, there are merge conflicts now.
> 
> Can you please fix it?

@gsantner I will look into it in these days. Forgive me for leaving you hanging.

**Comment by gsantner** [comment]
no problem

**Comment by gsantner** [comment]
converted the PR to draft due inactivity

**Review by gsantner — APPROVED** [review]
Thanks I rebased your changes on my own upon master...hope I did it all right and your intended changes still work.

I could at least press the >attach image< button, the text was inserted and a picture displayed in view mode. So should be OK I guess.

Thanks

**Comment by mehw** [comment]
> Thanks I rebased your changes on my own upon master...hope I did it all right and your intended changes still work.
> 
> I could at least press the >attach image< button, the text was inserted and a picture displayed in view mode. So should be OK I guess.
> 
> Thanks

Thank you very much for all your hard work.

Your consideration and merging my PR sure boost my mood ;)

Please, forgive my hiatus.  The last time I posted, I thought that I was ready to resume programming at full power but I had to accept to go easy limiting to what requires less brain effort.  I truly regret not being able to keep my promises.

On the bright side, I really enjoy programming and there are lots of things accumulated in my TODO list that keep me moving forward.

I treasure people, that is my strength, and my interest in debugging is an integral part of me.

I can't wait to be in tip-top shape for exploring the code.

**Comment by gsantner** [comment]
Thanks @mehw , feel welcome any time you want to join coding again.

Would you mind to install the latest v2.12.5 update and check if I did alright with the merge conflict? (So your initial change now works as you wanted it to)

File: app/src/main/java/net/gsantner/markor/frontend/AttachLinkOrFileDialog.java [hunk 1]
```diff
@@ -72,6 +74,9 @@ private static String getLinkFormat(final int textFormatId) {
     }
 
     private static String getAudioFormat(final int textFormatId) {
+        if (textFormatId == FormatRegistry.FORMAT_WIKITEXT) {
+            return "[[LINK|TITLE]]";
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/frontend/AttachLinkOrFileDialog.java:78]: for reference, this should have been `[[%LINK%|%TITLE%]]` :)

**Pull request merged**